### DISCUSSION TITLE
How to get string for a specific single key in desired locale?

### DISCUSSION THREAD

**Discussion by sahruday** [B3nedikt/restring#143] [General]
Is there a way to ignore the current locale present and use the default or the specified locale for a particular key?

There is a case based on a flag the specified string can or cannot be translated.

**Comment by B3nedikt** [comment]
Yeah, you can just use the repository directly, something like this:
```kotlin
Restring.stringRepository[Locale.ITALIAN]["the_string_key"]
```
would give you a string for the italian locale with the key "the_string_key".

**Reply by sahruday** [reply]
This works fine if key is present.
But if key is not present, it should use android resources as fallback.

**Reply by B3nedikt** [reply]
So basically you want the normal android behaviour?
This works with restring exactly the same way as without it:

Just get a localized resources instance somehow:

```kotlin
    fun getLocalizedResources(context: Context, desiredLocale: Locale?): Resources {
        var conf: Configuration = context.resources.configuration
        conf = Configuration(conf)
        conf.setLocale(desiredLocale)
        val localizedContext = context.createConfigurationContext(conf)
        return localizedContext.resources
    }
```

And then just call `getString` as usual:

```kotlin
    getLocalizedResources(this, Locale.ITALIAN).getString(R.your_string_key)
```
